Assets/Scripts/LogicServices/AlgorithmService.cs

```diff
@@ -11,18 +12,29 @@ public class AlgorithmService
     {
         private readonly SelfBuild _selfBuild;
         private readonly Randomness _randomness;
+        private readonly BuildEffect _buildEffect;
 
         public AlgorithmService()
         {
             Random random = new Random();
             _selfBuild = new SelfBuild(random);
             _randomness = new Randomness(random);
+            _buildEffect = new BuildEffect(random);
         }
 
         public bool Decide(Trade trade,Tribe originator, Tribe targetCpu)
         {
-            if (_randomness.Calculate()) return !_selfBuild.Calculate(trade,targetCpu);
-            return _selfBuild.Calculate(trade,targetCpu);
+            //Original Decisions
+            bool selfBuildDecision = _selfBuild.Calculate(trade, targetCpu);
+            bool buildEffectDecision = _buildEffect.Calculate(trade, targetCpu, originator);
+            
+            //Randomise Decisions
+            selfBuildDecision = _randomness.Calculate() ? selfBuildDecision : !selfBuildDecision;

```

Comment:
Why is the "randomness chance" the same? Shouldn't it be configurable based on the decision it used for?